pkg/registry/core/service/storage/rest.go

```diff
@@ -489,9 +485,6 @@ func (rs *REST) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 		return nil, false, err
 	}
 	externalTrafficPolicyUpdate(oldService, service)
```

Comment:
This is mutating the service to set `service.Spec.ExternalTrafficPolicy`, 
I think that is the reason why there is a validation here,  should `externalTrafficPolicyUpdate()` be moved to `strategy.PrepareForUpdate()`?

Reply:
good catch!

Reply:
made a new commit just for this

Reply:
Put new commit at the front - it can be merged independent of the rest.